src/path.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+'use strict';
+
+import { JsonObject } from 'cockpit';
+import { StrOrBytes, join_data } from 'cockpit/_internal/common.ts';
+import { Channel, ChannelPayload, ChannelOptions, BaseChannelOptions } from 'cockpit/channel.ts';
+
+interface PathOptions extends JsonObject {
+    // HACK: this one is also "undefined"? I guess
+    // superuser?: BaseChannelOptions['superuser'];
+    superuser?: "try" | "require";
+}
+
+type ReadChannelOptions<P extends ChannelPayload> = ChannelOptions<P> & {
+    path?: string;
+    max_read_size?: number;
+}
+
+// To be expanded further in the future
+interface ReadOptions extends BaseChannelOptions {
+    // 16 * 1024 * 1024
+    max_read_size?: number // TODO: default
+    binary?: boolean
+}
```

Comment:
So we have ReadOptions and ReadChannelOptions.... WHY! (This is seriously a bit silly)